.gitlab/ci/build_quartz.yml

```diff
@@ -0,0 +1,109 @@
+##############################################################################
+# Copyright (c) 2016-19, Lawrence Livermore National Security, LLC and Umpire
+# project contributors. See the COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (MIT)
+##############################################################################
+
+####
+# This is the share configuration of jobs for quartz
+
+####
+# In pre-build phase, allocate a node for builds
+allocate_resources_build_quartz:
+  tags:
+    - shell
+    - quartz
+  stage: allocate_resources
+  script:
+    - salloc -N 1 -c 36 -p pdebug -t 10 --no-shell --job-name=${BUILD_QUARTZ_ALLOC_NAME}
+  except:
+    - /_qnone/
+
+####
+# In post-build phase, deallocate resources
+# Note : make sure this is run even on build phase failure
+release_resources_build_quartz:
+  tags:
+    - shell
+    - quartz
+  stage: release_resources
+  script:
+    - scancel $(squeue -h --name=${BUILD_QUARTZ_ALLOC_NAME} --format=%A)
+  when: always
+  except:
+    - /_qnone/
+
+####
+# Generic qwartz job, extending build script
+.build_quartz:
+  variables:
+  tags:
+    - shell
+    - quartz
+  stage: build
+  extends: .build_quartz_script
+  except:
+    - /_qnone/
+
+
+####
+# Here are all quartz build jobs
+
+build_quartz_clang_3_9_1:
+  variables:
+    COMPILER: "clang_3_9_1"
```

Comment:
Can the list of compilers be somehow kept in a common yml file?  It seems like these are currently duplicated among build_{machinename}.yml files.

Reply:
I should have explained this part of the configuration :
1. These parts of the build_{machinename}.yml files should be generated in the future.
2. As far as I know there is no way to factorize the compiler or machine name here :
  - No matrix mechanism at the moment in gitlab
  - Compiler could be specified by a variable, and job name is a variable, but I couldn’t extract a substring form it (no support for bash string processing)
3. I am not sure the naming here is permanent (I don’t think so actually)
4. Sharing some configuration between cluster would add another layer in the modular setup (.gitlab-ci.yml includes .gitlab/ci/build_{cluster}.yml which includes .gitlab/ci/{all_clusters.yml}). IMO it adds unnecessary complexity (at least at the moment) and exceptions will surely appear soon or later (why defining xl compilers for all clusters ?).

Please comment :)